File: clients/fsharp/Debugging/ColoredVertex.fs
#nowarn "0058"

namespace AiCup22
namespace AiCup22.Debugging

open AiCup22

/// Point + color
type ColoredVertex = {
    /// Position
    Position: Model.Vec2;
    /// Color
    Color: Debugging.Color;
} with

    /// Write ColoredVertex to writer
    member this.writeTo(writer: System.IO.BinaryWriter) =
        this.Position.writeTo writer
        this.Color.writeTo writer
        ()

    /// Read ColoredVertex from reader
    static member readFrom(reader: System.IO.BinaryReader) = {
        Position = Model.Vec2.readFrom reader;
        Color = Debugging.Color.readFrom reader;
    }

File: clients/csharp/Debugging/ColoredVertex.cs
namespace AiCup22.Debugging
{
    /// <summary>
    /// Point + color
    /// </summary>
    public struct ColoredVertex
    {
        /// <summary>
        /// Position
        /// </summary>
        public AiCup22.Model.Vec2 Position { get; set; }
        /// <summary>
        /// Color
        /// </summary>
        public AiCup22.Debugging.Color Color { get; set; }
    
        public ColoredVertex(AiCup22.Model.Vec2 position, AiCup22.Debugging.Color color)
        {
            this.Position = position;
            this.Color = color;
        }
    
        /// <summary> Read ColoredVertex from reader </summary>
        public static ColoredVertex ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new ColoredVertex();
            result.Position = AiCup22.Model.Vec2.ReadFrom(reader);
            result.Color = AiCup22.Debugging.Color.ReadFrom(reader);
            return result;
        }
    
        /// <summary> Write ColoredVertex to writer </summary>
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            Position.WriteTo(writer);
            Color.WriteTo(writer);
        }
    
        /// <summary> Get string representation of ColoredVertex </summary>
        public override string ToString() {
            string stringResult = "ColoredVertex { ";
            stringResult += "Position: ";
            stringResult += Position.ToString();
            stringResult += ", ";
            stringResult += "Color: ";
            stringResult += Color.ToString();
            stringResult += " }";
            return stringResult;
        }
    }
}